ql/src/java/org/apache/hadoop/hive/ql/exec/GroupByOperator.java

```diff
@@ -389,14 +390,14 @@ protected void initializeOp(Configuration hconf) throws HiveException {
     isLlap = LlapDaemonInfo.INSTANCE.isLlap();
     numExecutors = isLlap ? LlapDaemonInfo.INSTANCE.getNumExecutors() : 1;
     firstRow = true;
+    memoryMXBean = ManagementFactory.getMemoryMXBean();
+    maxMemory = isTez ? getConf().getMaxMemoryAvailable() : memoryMXBean.getHeapMemoryUsage().getMax();
```

Comment:
is that safe to use `memoryMXBean.getHeapMemoryUsage().getMax()`?
````
Returns the maximum amount of memory in bytes that can be used for memory management. This method returns -1 if the maximum memory size is undefined.
````
when `Runtime.getRuntime().maxMemory()`
````
Returns the maximum amount of memory that the Java virtual machine will attempt to use. If there is no inherent limit then the value Long.MAX_VALUE will be returned.
````